robottelo/products.py

```diff
@@ -629,6 +630,12 @@ class RHELAnsibleEngineRepository(GenericRHRepository):
     _key = PRODUCT_KEY_ANSIBLE_ENGINE
 
 
+class RHELServerExtras(GenericRHRepository):
```

Comment:
We're moving away from using this module in the near future as part of the helpers refactor. Not blocking this or, but note with @jyejare that you're adding on to the file he's refactoring now.

Reply:
Hmm, alright - so I've been messing around with not adding another base class here, but as far as I can see, I can't pass the key value to it any other way, given how the class structure works - If I pass the key into just a genericRHRepository constructor, it calls it's parent class, BaseRepository, before it does anything with the "Key' value

This then calls a distro setter, which causes the GenericRHRepository to call it's setter, failing, since the key value hasn't been set yet. All of this _before_ the check to set the key value to the one passed in is done. 

I can refactor this a bit to allow for that, but I'm not sure which is the right way to go - that, or add another class that will likely get refactored away soon.